.github/actions/pmd/action.yml

```diff
@@ -0,0 +1,46 @@
+name: "PMD"
+description: "Run PMD against changes in a pull request"
+inputs:
+  pmd-version:
+    description: The version of PMD to use
+    required: false
+    default: "6.55.0"
+  create-github-annotations:
+    description: |
+      Whether to create inline comments on the PR using GH Advanced Security. This is free for open source projects but
+      requires a license for private repos.
+    required: false
+    default: "true"
+  fail-on-new-issues:
+    description: |
+      When set to true this fails the build if the PR introduces more issues than it resolves.
+    required: false
+    default: "true"
+
+runs:
+  using: "composite"
+  steps:
+    - uses: actions/checkout@v3
+      env:
+        REF_TO_CHECKOUT: ${{ inputs.auto-commit == 'true' && github.head_ref || '' }}
+      with:
+        ref: ${{ env.REF_TO_CHECKOUT }}
+
+    - name: Download ruleset
+      run: curl https://raw.githubusercontent.com/AlfrescoLabs/pmd-ruleset/master/pmd-ruleset.xml -o ${{ github.action_path }}/pmd-ruleset.xml
```

Comment:
I'd assume it would be ideal to provide a configurable input to customize the branch we're pulling the ruleset from, in case we need to handle different rulesets depending on the project?

Reply:
Ultimately I thought we would get rid of the external dependency and move the default ruleset(s) into this project. Alternatively we could make it configurable as you suggest.

Reply:
I was thinking about the 2 approaches as well, and while in most cases we tend to aggregate all necessary configuration files here, I think that this could be a reasonable exception as **it makes it so that we always pull the latest rulesets no matter which version of alfresco-build-tool we're using**. Furthermore I think we can slightly better leverage git to handle multiple rulesets as separate branches in a separate repository, than multiple files in the same branch _(the latter would require more manual copy-pasting in case of global updates as opposed to just a set of pulling from master + solving eventual conflicts, at the same time the "knowledge gate" to update all rulesets would be definitely lower than having to acknowledge the existence of a separate repository with multiple branches in the very same scenario -- so there's this tradeoff)_.

I'm slightly in favor of keeping the rulesets in a separate repository if the general consensus is that we'd work slightly better with 1 single ruleset file per branch over multiple branches + we could really benefit from being able to skip upgrading N repositories PMD action's version just to pull a newer version of a ruleset. Ultimately if you can think of reasons why we'd work better a different way I'd be glad to change my opinion though :)

Reply:
you can alse have another `actions/checkout@v3` step for the ruleset specific repo, adding an input to change the ref (likely to what domenico is suggesting)

I think we will figure it out once we integrate it in more projects which is the best way to handle rulesets, for now the current external repo approach can be enough for me